src/login/logind-dbus.c

```diff
@@ -2210,35 +2266,63 @@ static int method_schedule_shutdown(sd_bus_message *message, void *userdata, sd_
 
         m->scheduled_shutdown_type = a;
         m->shutdown_dry_run = dry_run;
+        m->scheduled_shutdown_timeout = elapse;
 
-        if (m->nologin_timeout_source) {
-                r = sd_event_source_set_time(m->nologin_timeout_source, nologin_timeout_usec(elapse));
+        r = manager_setup_shutdown_timers(m);
+        if (r < 0)
+                return r;
+
+        r = setup_wall_message_timer(m, message);
+        if (r < 0) {
+                m->scheduled_shutdown_timeout_source = sd_event_source_unref(m->scheduled_shutdown_timeout_source);
```

Comment:
`sd_event_source_disable_unref()`

Is it ok to leave `nologin_timeout_source` as is?

Reply:
was that way before but probably not :-) will fix